### PR TITLE
[python] prepare for PyPI

### PR DESCRIPTION
**Pull request by guolinke**

### CODE REVIEW

**Comment by henry0312** [comment]
type: PyPI (pypi), not pipy :)

**Comment by henry0312** [comment]
I can do リファクタリング `setup.py` after merge.

**Comment by guolinke** [comment]
@henry0312 can you refactor it base on this branch, then creat a PR to that branch?

**Comment by henry0312** [comment]
@guolinke yes, I can, but I'll have been busy until the end of June.
So I think It's better to merge this PR for now.

**Comment by guolinke** [comment]
@wxchan any comments?

**Comment by wxchan** [comment]
Is it possible to remove `--precompile` and infer whether it's precompiled? like run `find_packages` first and see if it's an exception?

**Comment by wxchan** [comment]
MANIFEST.in can be simplified a little:
```
warning: no files found matching '*.md'
warning: no files found matching '*' under directory 'include'
warning: no files found matching '*' under directory 'src'
warning: no previously-included files matching '*.o' found anywhere in distribution
warning: no previously-included files matching '*.obj' found anywhere in distribution
warning: no previously-included files matching '*.a' found anywhere in distribution
warning: no previously-included files matching '*.pyo' found anywhere in distribution
warning: no previously-included files matching '*.pyc' found anywhere in distribution
```

build_pip.py seems not necessary? we can run `python setup.py sdist` directly

**Comment by guolinke** [comment]
@wxchan 
I think the default option should be re-compile since it is more safe if update the cpp codes.

**Comment by wxchan** [comment]
ok, hope it wouldn't be confused to old users.

what is the getopt error you talked about yesterday?

**Comment by guolinke** [comment]
@wxchan  fixed by adding this line: https://github.com/Microsoft/LightGBM/blob/ddcee1e011336d68046cde257589da2eb730a9bf/python-package/setup.py#L29

**Comment by wxchan** [comment]
ok, no other issue.

**Comment by guolinke** [comment]
@wxchan 
after build the package, i try to install it via ``` pip install lightgbm-0.2a0.tar.gz```. 
But it shows a strange error : ``` No files/directories in C:\Users\xxx\AppData\Local\Temp\pip-vfzusp1m-build\pip-egg-info (from PKG-INFO)  ```

Do you have any idea about this ?

File: python-package/setup.py [hunk 1]
```diff
@@ -3,34 +3,96 @@
...
+        for opt, arg in opts:
+            if opt in ('-m', '--mingw'):
+                use_mingw = True
+            elif opt in ('-g', '--gpu'):
+                use_gpu = True
+            elif opt in ('-p', '--precompile'):
+                use_precompile = True
+    except getopt.GetoptError as err:
+        pass
+    sys.argv = sys.argv[0:2]
```
Comment (wxchan) [inline python-package/setup.py]: I got the same error on my mac, fix by removing this line

**Pull request merged**